src/win/tty.c

```diff
@@ -170,14 +170,14 @@ void uv__console_init(void) {
                                        0);
   if (uv__tty_console_handle != INVALID_HANDLE_VALUE) {
     CONSOLE_SCREEN_BUFFER_INFO sb_info;
-    QueueUserWorkItem(uv__tty_console_resize_message_loop_thread,
-                      NULL,
-                      WT_EXECUTELONGFUNCTION);
     uv_mutex_init(&uv__tty_console_resize_mutex);
     if (GetConsoleScreenBufferInfo(uv__tty_console_handle, &sb_info)) {
       uv__tty_console_width = sb_info.dwSize.X;
       uv__tty_console_height = sb_info.srWindow.Bottom - sb_info.srWindow.Top + 1;
     }
+    QueueUserWorkItem(uv__tty_console_resize_message_loop_thread,
+                      NULL,
+                      WT_EXECUTELONGFUNCTION);
```

Comment:
Not a Windows programmer, maybe the synchronization guarantees for QueueUserWorkItem() are such that it isn't necessary, but adding a critical section looks more Obviously Correct.
```suggestion
    uv_mutex_init(&uv__tty_console_resize_mutex);
    if (GetConsoleScreenBufferInfo(uv__tty_console_handle, &sb_info)) {
      uv_mutex_lock(&uv__tty_console_resize_mutex);
      uv__tty_console_width = sb_info.dwSize.X;
      uv__tty_console_height = sb_info.srWindow.Bottom - sb_info.srWindow.Top + 1;
      uv_mutex_unlock(&uv__tty_console_resize_mutex);
    }
    QueueUserWorkItem(uv__tty_console_resize_message_loop_thread,
                      NULL,
                      WT_EXECUTELONGFUNCTION);
```
There's an assert around line 2426 that looks like it can (still?) be hit now:
```c
assert(uv__tty_console_width != -1 && uv__tty_console_height != -1);
```
I'd simply remove it.

Reply:
Sure, that's valid.  I was aiming for the least-disruptive change.

Reply:
I removed the assert (agreed it's unnecessary), but left the initialization order as it is.  It doesn't make sense to acquire a mutex in the same function that initializes the mutex.

Reply:
> It doesn't make sense to acquire a mutex in the same function that initializes the mutex.

That's not the point I'm making. Wrapping the assignments in a critical section ensures other threads see the assignments, and in the right order.

QueueUserWorkItem() probably issues some kind of memory barrier but why take chances? This isn't performance-critical code.

Reply:
Ahh, I see what you mean.  On Windows, there is definitely an implied memory fence between all operations that start threads (or their equivalents, like QueueUserWorkItem), and the body of the thread.  Without that implied barrier, you would need to manually insert barriers any time you created a context object that is passed to CreateThread, QueueUserWorkItem, etc.